### PR TITLE
Convert all independent GET requests to RQ

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

File: studio/components/interfaces/BillingV2/Invoices.tsx [hunk 1]
```diff
@@ -1,92 +1,57 @@
...
       })
     }
   }
 
   return (
     <div className="container my-4 max-w-4xl space-y-1">
-      <Loading active={loading}>
+      {isLoading && <GenericSkeletonLoader />}
+
+      {isError && <AlertError error={error} subject="Failed to retrieve project invoices" />}
```
Comment (ivasilov) [inline studio/components/interfaces/BillingV2/Invoices.tsx:52]: We should use the new alerts here, right?

File: studio/components/interfaces/BillingV2/Invoices.tsx [hunk 2]
```diff
@@ -103,7 +68,7 @@ const Invoices = ({ projectRef }: InvoicesProps) => {
               <Table.tr>
                 <Table.td colSpan={6} className="p-3 py-12 text-center">
                   <p className="text-scale-1000">
-                    {loading ? 'Checking for invoices' : 'No invoices for this project yet'}
+                    {isLoading ? 'Checking for invoices' : 'No invoices for this project yet'}
```
Comment (ivasilov) [inline studio/components/interfaces/BillingV2/Invoices.tsx:71]: `isLoading` and `isSuccess` can't both be true, right?

File: studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddress.tsx [hunk 3]
```diff
@@ -54,15 +61,15 @@ const BillingAddress = ({ loading, address, onAddressUpdated }: BillingAddressPr
   const onSubmit = async (values: any, { setSubmitting }: any) => {
     try {
       setSubmitting(true)
-      const updatedCustomer = await patch(`${API_URL}/organizations/${orgSlug}/customer`, {
+      const updatedCustomer = await patch(`${API_URL}/organizations/${slug}/customer`, {
```
Comment (ivasilov) [inline studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddress.tsx:64]: This refactor would've been unnecessary once we have the this PATCH call in a mutation, right? It will invalidate the cache key and the GET query will happen on its own.

File: studio/components/interfaces/BillingV2/Invoices.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/BillingV2/Invoices.tsx:52]: Ahh yup! AlertError is already using the new Alert component - its just a standardized component on how we want to show UI Alerts across the dashboard - an error message and a quick button to direct users to the support form the title prefilled and error message added to the message body

File: studio/components/interfaces/BillingV2/Invoices.tsx (hunk 2 above)
Comment (joshenlim) [reply studio/components/interfaces/BillingV2/Invoices.tsx:71]: nope - the states from RQ are all mutually exclusive (isLoading, isSuccess, isError for the one's that we commonly use, and all the other states too)

File: studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddress.tsx (hunk 3 above)
Comment (joshenlim) [reply studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddress.tsx:64]: yup! the PATCH call mutation change is added in https://github.com/supabase/supabase/pull/16347/files

File: studio/components/interfaces/BillingV2/Invoices.tsx (hunk 2 above)
Comment (joshenlim) [reply studio/components/interfaces/BillingV2/Invoices.tsx:71]: ohh sorry Alaister just pointed out that isLoading will never be true here since its within the isSuccess conditional - will update! 🙏

File: studio/data/invoices/project-invoices-query.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+export type ProjectInvoicesError = ResponseError
+
+export const useProjectInvoicesQuery = <TData = ProjectInvoicesData>(
+  { projectRef, offset, limit }: ProjectInvoicesVariables,
+  {
+    enabled = true,
+    ...options
+  }: UseQueryOptions<ProjectInvoicesData, ProjectInvoicesError, TData> = {}
+) =>
+  useQuery<ProjectInvoicesData, ProjectInvoicesError, TData>(
```
Comment (alaister) [inline studio/data/invoices/project-invoices-query.ts:40]: For the future we could look at making this `useInfiniteQuery` since it takes a limit/offset.

File: studio/data/invoices/project-invoices-query.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query'
+import { get } from 'data/fetchers'
+import { ResponseError } from 'types'
+import { invoicesKeys } from './keys'
+
+export type ProjectInvoicesVariables = {
+  projectRef?: string
+  offset: number
+  limit: number
```
Comment (alaister) [inline studio/data/invoices/project-invoices-query.ts:9]: Seems strange to me that these are required – can't we provide defaults? e.g. `offset = 0` and `limit = 50`?

File: studio/data/invoices/project-invoices-query.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+) =>
+  useQuery<ProjectInvoicesData, ProjectInvoicesError, TData>(
+    invoicesKeys.projectInvoices(projectRef),
+    ({ signal }) => getProjectInvoices({ projectRef, offset, limit }, signal),
+    {
+      enabled:
+        enabled &&
+        typeof projectRef !== 'undefined' &&
+        typeof limit !== 'undefined' &&
+        typeof offset !== 'undefined',
```
Comment (alaister) [inline studio/data/invoices/project-invoices-query.ts:48]: If they weren't required these wouldn't be needed either

File: studio/data/invoices/invoices-query.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query'
+import { get } from 'data/fetchers'
+import { ResponseError } from 'types'
+import { invoicesKeys } from './keys'
+
+export type InvoicesVariables = {
+  customerId?: string
+  slug?: string
+  offset: number
+  limit: number
```
Comment (alaister) [inline studio/data/invoices/invoices-query.ts:10]: Same comment here about these being required

File: studio/data/fetchers.ts [hunk 8]
```diff
@@ -89,9 +89,11 @@ export const del: typeof _del = async (url, init) => {
 export const head: typeof _head = async (url, init) => {
   const headers = await constructHeaders(init?.headers)
 
+  // @ts-ignore
```
Comment (alaister) [inline studio/data/fetchers.ts:92]: Why is this needed here?

**Review by alaister — CHANGES_REQUESTED** [review]
Looking good! The pagination on invoices doesn't seem to be working, though. I think changing it to an infinite query might be the way to go

File: studio/data/fetchers.ts (hunk 8 above)
Comment (joshenlim) [reply studio/data/fetchers.ts:92]: ahh i'm only running into a ts error when i added the parseAs: 'text' - wondering if you might know whats the cause as i'm not really sure :/ 
image

File: studio/data/invoices/project-invoices-query.ts (hunk 5 above)
Comment (joshenlim) [reply studio/data/invoices/project-invoices-query.ts:9]: ahh yeap makes sense

File: studio/data/fetchers.ts (hunk 8 above)
Comment (alaister) [reply studio/data/fetchers.ts:92]: I think it's because it doesn't know the endpoint, and so it doesn't know if it supports `parseAs` yet. If you pass in `parseAt` at the call site instead of in this file, it shouldn't cause a type error.

**Review by alaister — APPROVED** [review]
LGTM!
